app/src/main/java/com/x8bit/bitwarden/data/platform/util/StringExtensions.kt

```diff
@@ -15,7 +15,16 @@ private const val ANDROID_APP_PROTOCOL: String = "androidapp://"
  */
 fun String.toUriOrNull(): URI? =
     try {
-        URI(this)
+        val uri = URI(this)
+        if (
+            uri.host == null &&
+            this.contains(".") &&
+            !this.hasHttpProtocol()
```

Comment:
What if it has a different protocol, like `androidapp`?

Reply:
in that case if it has a valid scheme then host would be present, the specific case something `foo.com:1000` when put in the `URI("foo.com:1000")` -> uri.host == null and uri.scheme == "foo.com", assuming its parsing for the first `:` to get the scheme.

Reply:
so it it has a valid scheme and a host we would be presuming that `uri.host != null` and the `URI()` will parse it correctly.

Reply:
did a little changing of where we do this check to only be when grabbing the host, also added tests